sublime_jedi/go_to.py

```diff
@@ -151,11 +157,144 @@ def prepare_option(self, option):
+
+        def handle_rename(new_name: str) -> None:
+            groups = []  # type: List[List[Tuple[str, int, int]]]
+            files = set()  # type: Set[str]
+
+            for option in options:
+                file = option[0]
+                if not file:  # can't replace text (or even show usages) in unsaved file
+                    continue
+                if file in files:
+                    groups[-1].append(option)
+                else:
+                    groups.append([option])
+                files.add(file)
+
+            for group in groups:
+                rename_in_file(group, group[0][0], new_name)
+
+        def rename_in_file(group, file, new_name):
+            # type: (List[Tuple[str, int, int]], str, str) -> None
+            with open(file) as f:
+                text = f.read()
+            original_text = text
+            offset = 0
+
+            for option in group:
+                assert text and name
+                _, row, col = option
+                point = text_point(original_text, row-1, col-1)
+
+                text = text[:point + offset] + new_name + text[point + offset + len(name):]
+                offset += len(new_name) - len(name)
+
+            with open(file, "w") as f:
+                f.write(text)
+
+        def handle_choose(idx):
+            if not name:
+                return
+            if idx == 0:
+                view.window().show_input_panel("New name:", name, handle_rename, None, None)
+                return
+            self._jump_to_in_window(idx - 1 if idx != -1 else idx)
+
+        def handle_highlight(idx):
+            if idx == 0:
+                return
+            partial(self._jump_to_in_window, transient=True)(idx - 1 if idx != -1 else idx)
+
+        # Show the user a selection of filenames
+        files = set()  # type: Set[str]
+        for option in options:
+            files.add(option[0])
+        first_option = [[
+            "rename " + '"{}"'.format(name),
+            "{} occurrence{} in {} file{}".format(
+                len(options), 's' if len(options) != 1 else '', len(files), 's' if len(files) != 1 else '')
+        ]]
+        active_window.show_quick_panel(
+            first_option + [self.prepare_option(o) for o in options],
+            handle_choose,
+            on_highlight=handle_highlight)
+        self._window_quick_panel_open_window(view, options)
 
     def prepare_option(self, option):
         return [to_relative_path(option[0]),
                 "line: %d column: %d" % (option[1], option[2])]
+
+
+def expand_selection(view, point):
+    # type: (Any, Any) -> str
+    name = ""
+    _, col = view.rowcol(point.begin())
+    for match in re.finditer(r"[A-Za-z0-9_]+", view.substr(view.line(point.begin()))):
+        if match.start() <= col and match.end() >= col:
+            name = match.group()
+    return name
+
+
+def text_point(text: str, row: int, col: int) -> int:
```

Comment:
could you please add a short description what is going on here?

Reply:
Yeah, for sure